src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -331,6 +332,42 @@ public function move($path, $target)
         return rename($path, $target);
     }
 
+    /**
+     * Create a file from a base64 image
+     *
+     * @param  string  $base64File
+     * @return string
+     */
+    public function fromBase64(string $base64File)
+    {
+        if (!$this->isBase64File($base64File)) {
+            throw new InvalidBase64FileException();
+        }
+
+        if (strpos($base64File, 'data:image') !== false) {
+            $base64File = explode(',', $base64File, 2)[1];
+        }
+
+        $file_data = base64_decode($base64File);
+        return $file_data;
+    }
+
+    /**
+     * Check if base64 is valid
+     *
+     * @param  string  $base64File
+     * @return bool
+     */
+    public function isBase64File(string $base64)
+    {
+        if (strpos($base64, ';base64,') !== false) {
+            $base64 = explode(';base64,', $base64)[1];
```

Comment:
There's another optional segment: media type parameters like `charset`

Reply:
If i get you correctly there are base64 files that might also have media parameter types like, for example "charset=application/octet-stream"

Reply:
`charset` would be more like `"UTF-8"` or `"ISO-8859-1"`. `"application/octet-stream"` would be used like this if I'm not mistaken: `data:application/octet-stream;charset=UTF-8;page=21,the%20data:1234,5678`

Reply:
Okay let me get to it